fe/topology.py

```diff
@@ -690,23 +697,38 @@ class DualTopologyStandardDecoupling(DualTopology):
     Furthermore, the ligand's non-ring torsions are turned off at the standard state to improve
     sampling.
 
+    Blocker molecule has all of its torsions turned off throughout
+
     """
-    def parameterize_proper_torsion(self, ff_params):
+    def parameterize_proper_torsion(self, ff_params, core_idxs_a, core_idxs_b):
         # (ytz): TBD need to do this for Base Topology as well
 
         torsion_params, torsion_potential = super().parameterize_proper_torsion(ff_params)
 
-        mol_c = Chem.CombineMols(self.mol_a, self.mol_b)
-        membership = get_ring_membership(mol_c)
+        # mol_c = Chem.CombineMols(self.mol_a, self.mol_b)
+        membership_a = get_ring_membership(self.mol_a)
+        membership_b = get_ring_membership(self.mol_b)
 
         num_torsions = torsion_params.shape[0]
 
         lambda_mult_idxs = np.zeros(num_torsions, dtype=np.int32)
         lambda_offset_idxs = np.ones(num_torsions, dtype=np.int32)
 
-        for torsion_idx, (_, b, c, _) in enumerate(torsion_potential.get_idxs()):
-            if membership[b] != membership[c]:
-                lambda_offset_idxs[torsion_idx] = 0
+        NA = self.mol_a.GetNumAtoms()
+
+        for torsion_idx, (i, b, c, j) in enumerate(torsion_potential.get_idxs()):
+
+            # does this torsion belong to mol_a?
+            if b < NA and c < NA and i < NA and j < NA:
+                if (membership_a[b] != membership_a[c]) and (b in core_idxs_a and c in core_idxs_a):
+                    lambda_offset_idxs[torsion_idx] = 0
+            # are we in in b?
+            if b > NA and c > NA and i > NA and j > NA:
```

Comment:
Should this be `b >= NA`? If I understand this, shouldn't the first atom of Mol B be `idx == NA`?

Reply:
I think you were looking at a stale/outdated page - this was updated last night

Reply:
You are right, looks good now!